### PR TITLE
port the MHD PLM reconstruction to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
all tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-05-22-008/index.html

File: Source/mhd/Castro_mhd.H [hunk 1]
```diff
@@ -1,10 +1,50 @@
...
+        amrex::Array4<amrex::Real const> const& Bx,
+        amrex::Array4<amrex::Real const> const& By,
+        amrex::Array4<amrex::Real const> const& Bz,
+        amrex::Array4<amrex::Real> const& qleft,
+        amrex::Array4<amrex::Real> const& qright,
+        amrex::Array4<amrex::Real const> const& srcQ,
+        const amrex::Real dt);
+
+    void
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
```
Comment (maxpkatz) [inline Source/mhd/Castro_mhd.H:24]: These qualifiers need to go before the function type

File: Source/mhd/mhd_eigen.H [hunk 2]
```diff
@@ -0,0 +1,622 @@
+#ifndef CASTRO_MHD_EIGEN_H
+#define CASTRO_MHD_EIGEN_H
+
+#include "mhd_util.H"
+
+void
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+Castro::evals(Real* lam, const Real as_in, Real* Q, const int dir) {
```
Comment (maxpkatz) [inline Source/mhd/mhd_eigen.H:8]: Can we use Array1D for things like `lam` and `Q` rather than raw pointers?

File: Source/mhd/Castro_mhd.H (hunk 1 above)
Comment (zingale) [reply Source/mhd/Castro_mhd.H:24]: should be fixed now

File: Source/mhd/mhd_eigen.H (hunk 2 above)
Comment (zingale) [reply Source/mhd/mhd_eigen.H]: done

**Pull request merged**